tests/rptest/tests/upgrade_test.py

```diff
@@ -146,3 +147,39 @@ def test_rolling_upgrade(self):
         post_upgrade_num_msgs = self.producer.num_acked
         self.run_validation(min_records=post_upgrade_num_msgs +
                             (self.producer_msgs_per_sec * 3))
+
+    @cluster(num_nodes=5, log_allow_list=RESTART_LOG_ALLOW_LIST)
+    @parametrize(upgrade_after_rollback=True)
+    @parametrize(upgrade_after_rollback=False)
+    def test_rolling_upgrade_with_rollback(self, upgrade_after_rollback):
+        self.installer.install(self.redpanda.nodes, RedpandaInstaller.HEAD)
+
+        # Upgrade one node.
+        first_node = self.redpanda.nodes[0]
```

Comment:
some code duplication maybe we can pack these into a single lambda and call w/ params?

Reply:
Done